src/rcdom_with_line_numbers.rs

```diff
@@ -0,0 +1,179 @@
+// This provides a wrapper around RcDom which tracks line numbers in the errors.
+
+use html5ever::interface::TreeSink;
+use html5ever::{
+    tendril::StrTendril,
+    tree_builder::{ElementFlags, NextParserState, NodeOrText, QuirksMode},
+    Attribute, ExpandedName, QualName,
+};
+use markup5ever_rcdom::{Handle, RcDom};
+use std::borrow::Cow;
+use std::io;
+
+pub struct RcDomWithLineNumbers {
+    dom: RcDom,
+    current_line: u64,
+}
+
+impl RcDomWithLineNumbers {
+    // Expose out the document and errors from the inner RcDom
+    pub fn document(&self) -> &Handle {
+        &self.dom.document
+    }
```

Comment:
I don't like how this makes `RcDomWithLineNumbers` have a different public API than `RcDom`. Also sometimes I need to call `clone()` on the result of this, and sometimes not. Probably there is something better I could do here.